lib/streamlit/elements/widgets/button_group.py

```diff
@@ -192,7 +223,7 @@ def feedback(
         self,
         options: Literal["thumbs", "faces", "stars"] = "thumbs",
         *,
-        key: str | None = None,
+        key: str | int | None = None,
```

Comment:
nit: would `Key | None = None` work?
(saw we're using it for `_internal_button_group`)

Reply:
it would work and admittedly we are not super consisting here, but I think we _try_ to not use internal types for the public facing API, because we don't expose the types nicely to users today. So when we use `Key` they see `Key`, but they have to dig into the code to know what `Key` is and they cannot really nicely use the type for their own code. So here we willingly violate the DRY principle for better UX. Here is also another comment in the st.audio PR related to this: https://github.com/streamlit/streamlit/pull/9404#discussion_r1746983800
Does this make sense to you?

Reply:
That's great context and agree this -- I just didn't realize they sit in different layers of the APIs

Reply:
Btw. the expose typings project is [here](https://www.notion.so/snowflake-corp/Expose-type-definitions-in-streamlit-typing-8aebac0560714c5a97d99793bd4972c2), and I added a note to review `Key` and other types with use in our API.

Reply:
alright, I have updated it 👍